### ISSUE TITLE
Constructor with problems build updated guardtower

### ISSUE THREAD

**Issue by scriptsengineer** [ldtteam/minecolonies#1459]
### Minecolonies version

version 1.12.1-0.8.4000

### Expected behavior

- Constructor destroy slab block to updated guardtower3

### Actual behaviour

- Builder receives slabs of wood infinitely, one every second, without removing from the guard tower of level 2

### Steps to reproduce the problem

1. Create colony in an earlier version
2. Update version
3. create an upgrade requisition for the guard tower from level 2 to level 3
Note:No error occurred in other constructions.

**Comment by Asherslab** [comment]
Is this the Guard Tower in the Barracks or the stand-alone guard tower?

**Comment by scriptsengineer** [comment]
I have not yet built the barracks, it was a stand-alone guard tower

**Comment by scriptsengineer** [comment]
I ended up correcting the problem myself by removing the slab blocks from the level 2 tower, only the slab block already caused the construction to return to normal

**Comment by Asherslab** [comment]
Strange.. strange @Raycoms Any ideas? otherwise i guess we close this? seems kinda like a fluke/ really weird occurrence to me.

**Comment by scriptsengineer** [comment]
I think it might be a coincidence that I use the same world in several updates. Another thing is that there is no error in any block other than this one.

**Comment by Asherslab** [comment]
yeah, it's quite weird. i'd think it's probably a fluke. if not we'll hear about it again lol

**Comment by scriptsengineer** [comment]
Thanks for the attention @Asherslab, anyway.

**Comment by scriptsengineer** [comment]
I must point out that the problem keeps happening, I will report that now at the moment the problem was with oak for upgrading from level 1 to level 2.

**Comment by Raycoms** [comment]
is anything printed to the log?

2017-09-22 19:55 GMT+01:00 Rafael Correa <notifications@github.com>:

**Comment by scriptsengineer** [comment]
I realized now that this happens over time a colonist on top of the change block. Nothing found in log

**Comment by scriptsengineer** [comment]
I put the guard's position on top of the tower, right on top of a slab, when the builder will upgrade the block below and next he just gets the blocks infinitely. Note: The block below the guard is removed and the guard falls directly below. the next block is what bug.

I've been checking times, to know which block they bug. 

Thanks @Raycoms  for attention.

**Comment by Raycoms** [comment]
Wow that seems like an extremely interesting bug =O I will take a look at
it.

2017-09-23 14:59 GMT+01:00 Rafael Correa <notifications@github.com>:

**Comment by Raycoms** [comment]
does this only happen in 1.12 or can you reproduce that in 1.11 or 1.10 as
well?

2017-09-23 15:00 GMT+01:00 Ray Neiheiser <ray.neiheiser@gmail.com>:

**Comment by scriptsengineer** [comment]
The bug started after the 4000+ version, previous ones did not happen, and by what you perceive happens with transparency blocks (slab, stairs, etc.)
unfortunately at the moment I can not play to test, but in the next few days test in different versions

**Comment by ogradyjd** [comment]
Bug is happening in 1.10 as well (minecolonies-universal-1.10.2-0.8.4554.jar).  It's happening for me in trying to get my barracks upgraded from 4 to 5.  It happened when I upgraded from 3 to 4, but I fixed it by destroying the barracks by hand and then re-issuing the upgrade request.  With 4 to 5, I have yet to get the builder to stop getting endless spruce slabs.  He sits there on the corner of the barracks and his inventory just fills with slabs, with nothing ever removed from the building.
